packages/recommend-react/src/useStableValue.ts

```diff
@@ -0,0 +1,15 @@
+import { dequal } from 'dequal/lite';
+import { useEffect, useState } from 'react';
+
+export function useStableValue<TValue>(value: TValue) {
+  const [stableValue, setStableValue] = useState<TValue>(() => value);
+
+  useEffect(() => {
+    if (dequal(stableValue, value) === false) {
+      setStableValue(value);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
this is required because stableValue is not considered "safe" by react right? I wonder if a comment is needed?

Reply:
Indeed, I've added a comment https://github.com/algolia/recommend/pull/52/commits/2c3479e1449111e36298c278b7be35771ceebb96